src/states/common/approved.rs

```diff
@@ -212,7 +221,10 @@ pub trait Approved: Base {
                 }
 
                 let rand_index = self.rng().gen_range(0, p2p_recipients.len());
-                (version, p2p_recipients[rand_index].clone())
+                let recipient = p2p_recipients[rand_index].clone();
+                let version = self.get_parsec_version_for(recipient.public_id());
```

Comment:
No sure about that will come back to it.

Reply:
it does not quite fit with the poke thing, but probably ok now that we update the version with the poke message so that is the version we will get, and I guess with the removal of member version for elder, we get the latest version always which is what we want (i.e only send response to older version not request, except to adults that poke us).

Reply:
So I'm still not clear how that fit in this PR:
 -> If you get a MemberKnowledge, we are going to be in the `Some((v, p)) => (v, p),` branch.
 -> If you don't in what circumstance do you need to send not the latest Parsec gossip? Did it change with this PR?

Reply:
Hm, right. I guess this only makes sense if we went with the non-periodic message approach. Will revert it.